travis/git_run.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+
+import subprocess
+
+
+class GitRun(object):
+    def __init__(self, repo_path):
+        self.repo_path = repo_path
+
+    def run(self, command):
+        """Execute git command in bash
+        :param list command: Git cmd to execute in self.repo_path
+        :return: String output of command executed.
+        """
+        cmd = ['git', '--git-dir=' + self.repo_path] + command
+        try:
+            res = subprocess.check_output(cmd)
+        except subprocess.CalledProcessError:
+            res = None
+        if isinstance(res, basestring):
+            res = res.strip('\n')
+        return res
+
+    def get_items_changed(self, base_ref='HEAD'):
+        """Get name of items changed in self.repo_path
+        This is a wrapper method of git command:
+            git diff-index --name-only --cached {base_ref}
+        :param base_ref: String of branch or sha base.
+            e.g. "master" or "SHA_NUMBER"
+        :return: List of name of items changed
+        """
+        command = ['diff-index', '--name-only',
+                   '--cached', base_ref]
+        res = self.run(command)
+        items = res.split('\n') if res else []
+        return items
```

Comment:
This is for use:
```python
git_obj = GitRun(REPO_PATH)
git_obj.run(['fetch'])
git_obj.run(['checkout'])
git_obj.get_items_changed('8.0')
```
(This code is similar to runbot.)

Other case, we will need to add the global values in each method call, example:
```python
git_run.run(['git', '--repo-path=' + REPO_PATH, 'fetch'])
git_run.run(['git', '--repo-path=' + REPO_PATH, 'checkout'])
git_run.get_items_changed(REPO_PATH, '8.0')
```

IMHO The first one is clearer.
What do you think?